File: Selenium/Activities/Activity12.java
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity12 {

    public static void main(String[] args) {

        ChromeDriver driver = new ChromeDriver();
        WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(10));
        
        driver.manage().window().maximize();

        driver.get("https://training-support.net/webelements/dynamic-controls");
        System.out.println("Title of page is : " + driver.getTitle());

        WebElement checkbox = driver.findElement(By.id("checkbox"));
        WebElement toggleCheckbox = driver.findElement(By.xpath("//button[text()='Toggle Checkbox']"));
        if(checkbox.isDisplayed()) {
        	System.out.println("checkbox is visible");
        }
        toggleCheckbox.click();
        wait.until(ExpectedConditions.invisibilityOf(checkbox));
        if (!(checkbox.isDisplayed())) {
        	System.out.println("checkbox is not visible");
        }
        toggleCheckbox.click();
        wait.until(ExpectedConditions.visibilityOf(checkbox));
        if(checkbox.isDisplayed()) {
        	System.out.println("checkbox is visible");
        	checkbox.click();
        }
     
        
        
        driver.quit();
    }
}

File: Selenium/Activities/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Chrome()
wait = WebDriverWait(driver, 10)

driver.maximize_window()
driver.get("https://training-support.net/webelements/dynamic-controls")
print("Title of page is :", driver.title)

checkbox = driver.find_element(By.ID, "checkbox")
toggle_checkbox = driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']")

if checkbox.is_displayed():
    print("checkbox is visible")

toggle_checkbox.click()
wait.until(EC.invisibility_of_element(checkbox))

if not checkbox.is_displayed():
    print("checkbox is not visible")

toggle_checkbox.click()
wait.until(EC.visibility_of(checkbox))

if checkbox.is_displayed():
    print("checkbox is visible")
    checkbox.click()

driver.quit()
